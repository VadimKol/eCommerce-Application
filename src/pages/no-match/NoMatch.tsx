import { useNavigate } from 'react-router-dom';

import { useAppStyles } from '@/hooks/useAppStyles';

export function NoMatch(): JSX.Element {
  const navigate = useNavigate();
  const appStyles = useAppStyles();

  return (
    <main className={appStyles.main}>
      <h1>404</h1>
      <p>Sorry, page not found.</p>
      <button type="button" onClick={() => navigate(-1)}>
        Previous page
      </button>
      <button type="button" onClick={() => navigate('/')}>
        Home page
      </button>
    </main>
  );
}
