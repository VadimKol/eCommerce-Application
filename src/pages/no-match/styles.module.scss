.main {
  padding: 20px;
}

.container {
  display: flex;
  flex-direction: column;
  max-width: 100%;
  margin: auto;
  padding: 20px 40px 30px;
  text-align: center;
  color: var(--primary-color);
  background:
    linear-gradient(var(--background-color) 0 0) padding-box,
    linear-gradient(45deg, var(--primary-color), var(--secondary-color)) border-box;
  border: 5px solid transparent;
  border-radius: 10px;
}

.title {
  margin-block: 0;
  font-family: var(--secondary-font);
  font-size: 8em;
  line-height: 1;
  text-shadow:
    0 0 10px var(--secondary-color),
    0 0 10px var(--secondary-color),
    0 0 20px var(--secondary-color);
  -webkit-text-fill-color: var(--background-color);
  -webkit-text-stroke-width: 1px;
  -webkit-text-stroke-color: var(--secondary-color);
}

.text {
  margin-block: 0 20px;
}

.buttons {
  display: flex;
  gap: 10px;
  justify-content: space-between;
}
