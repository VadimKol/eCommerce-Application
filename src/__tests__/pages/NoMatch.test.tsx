import '@testing-library/jest-dom';

import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

import { NoMatch } from '@/pages/no-match/NoMatch';

test('renders NoMatch component correctly', () => {
  const { asFragment } = render(
    <MemoryRouter>
      <NoMatch />
    </MemoryRouter>,
  );

  expect(asFragment()).toMatchSnapshot();
});
