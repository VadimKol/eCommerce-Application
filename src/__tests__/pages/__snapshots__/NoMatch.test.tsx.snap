// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders NoMatch component correctly 1`] = `
<DocumentFragment>
  <main
    class="main main"
  >
    <div
      class="container"
    >
      <h1
        class="title"
      >
        404
      </h1>
      <p
        class="text"
      >
        The page you requested was not found.
      </p>
      <div
        class="buttons"
      >
        <button
          class="button button--tertiary"
          id=""
          type="button"
        >
          Previous page
        </button>
        <button
          class="button button--tertiary"
          id=""
          type="button"
        >
          Home page
        </button>
      </div>
    </div>
  </main>
</DocumentFragment>
`;
